### DISCUSSION TITLE
Workspaces, Docker, and containerized packages?

### DISCUSSION THREAD

**Discussion by magnus-trent** [yarnpkg/berry#3268] [Q&A] [answered]
Hey everyone, 

I'm trying to see about using PnP in a monorepo, where the frontend and backend are containerized. However the cache is stored at the project root, and I'm **very ** new to Docker. Does anyone have an idea how this would be possible with Yarn Modern? Or Secondly, how would you recommend containerizing an application like this?

I was thinking, `packages/frontend` and `packages/backend` along with a `postgres` Docker image, but Yarn stores everything in the parent cache, making it hard to individually containerize the two services.

**Comment by gex** [comment, marked as answer]
i got this idea in #3201 and it will probably help you as well. there is a plugin for building an image for a workspace: [yarn-plugin-docker-build](https://github.com/Dcard/yarn-plugins/tree/master/packages/docker-build)

**Reply by magnus-trent** [reply]
Awesome! I can't thank you enough!!
